engine/classes/ElggEntity.php

```diff
@@ -1686,7 +1686,14 @@ protected function update() {
 		unset($persisted_entity);
 
 		if ($allow_edit) {
-			$allow_edit = _elgg_services()->events->trigger('update', $this->type, $this);
+			$allow_edit = elgg_trigger_before_event('update', $this->type, $this);
```

Comment:
I thought the before/after functions were there for just this reason

This doesn't really work without the after hook, as a plugin/something could prevent the update, but then the comments would get changed to a different access_id - making the situation worse.